tests/test_graph_processing.py

```diff
@@ -0,0 +1,116 @@
+"""
+Test file
+"""
+
+import pytest
+
+from power_system_simulation.graph_processing import (
+    GraphCycleError,
+    GraphNotFullyConnectedError,
+    GraphProcessor,
+    IDNotFoundError,
+    IDNotUniqueError,
+    InputLengthDoesNotMatchError,
+)
+
+
+class Tests:
+    """
+    Testing the raise exception in case of faulty graph
+    """
+
+    def test_id_not_unique_error_1(self):
+        """
+        Testing vertex_ids should be unique
+        """
+        vertex_ids = [1, 2, 3, 4, 4]
+        edge_ids = [1, 2, 3, 4]
+        edge_vertex_id_pairs = [(1, 2), (2, 3), (3, 4), (1, 3)]
+        edge_enabled = [1, 1, 1, 0]
+        source_vertex_id = 1
+        with pytest.raises(IDNotUniqueError):
+            GraphProcessor(vertex_ids, edge_ids, edge_vertex_id_pairs, edge_enabled, source_vertex_id)
```

Comment:
suggestion: This is entirely optional, but in our team we use the following structure in our tests to make them more readable in my opinion. You describe your "input" with GIVEN, what you do with WHEN and finally what you expect to happen with THEN. Feel free to keep it the way you have things, it's entirely a style thing 🙂
```suggestion
        # GIVEN
        # A list of vertex ids that are not unique
        vertex_ids = [1, 2, 3, 4, 4]
        edge_ids = [1, 2, 3, 4]
        edge_vertex_id_pairs = [(1, 2), (2, 3), (3, 4), (1, 3)]
        edge_enabled = [1, 1, 1, 0]
        source_vertex_id = 1
        
        # WHEN
        # The graph processor is initiated
        # THEN
        # It should raise an IDNotUniqueError
        with pytest.raises(IDNotUniqueError):
            GraphProcessor(vertex_ids, edge_ids, edge_vertex_id_pairs, edge_enabled, source_vertex_id)
```